dist_tables/dml.rst

```diff
@@ -38,6 +38,70 @@ For example:
+
+We can create a rollup table :code:`daily_page_views` to hold these results and fill it with the results of the previous query:
+
+.. code-block:: postgres
+
+  CREATE TABLE daily_page_views (
+    day date,
+    page_id int,
+    view_count bigint,
+    primary key (day, page_id)
+  );
+
+  INSERT INTO daily_page_views (day, page_id, view_count)
+    -- this is our original select query
+    SELECT view_time::date AS day, page_id, count(*) AS view_count
+    FROM page_views
+    GROUP BY view_time::date, page_id;
+
+We can use PostgreSQL's upsert feature to incrementally update certain parts of the rollup table. For instance suppose we have already rolled up visits happening before a certain timestamp (we'll call it :code:`$1`) and want to update the rollups to include more recent views. Just add a WHERE clause and specify "ON CONFLICT." The latter is PostgreSQL's technique for doing `upserts <https://www.postgresql.org/docs/9.5/static/sql-insert.html#SQL-ON-CONFLICT>`_.
+
+.. code-block:: postgres
+
+  INSERT INTO daily_page_views (day, page_id, view_count)
+    SELECT view_time::date AS day, page_id, count(*) AS view_count
+    FROM page_views
+    WHERE view_time >= $1
+    GROUP BY view_time::date, page_id
+    ON CONFLICT (day, page_id) DO UPDATE SET
+      view_count = daily_page_views.view_count + EXCLUDED.view_count;
+
+These are the basics of using insert into select on a single node database. Using it effectively in Citus for real-time analytics requires more understanding of Citus' distributed query execution. Citus implements insert into select by pushing down the select query to each shard. When the destination table for inserted values is distributed by an analogous column as the table being selected then the inserting and selecting can be "co-located" together on each shard. This minimizes network traffic between nodes and allows highly parallel execution.
+
+For fast execution ensure that:
+
+- The tables queried and inserted are distributed on analogous columns
+- The select query includes the distribution column
+- The insert statement includes the distribution column value
+- GROUP BY includes the distribution column
+
+The example query shown above will work well if :code:`page_views` and :code:`daily_page_views` are distributed by :code:`page_id` column. Distributing our data by page allows rollup to scale linearly with the number of cores.
+
+.. code-block:: postgres
+
+  SELECT create_distributed_table('page_views', 'page_id');
+  SELECT create_distributed_table('daily_page_views', 'page_id');
+
+There are a few things to be aware of when running distributed insert into selects. First, Citus takes locks to ensure consistency when the :ref:`replication_factor` is greater than one. In this situation if you know your inserts are not going to affect an ongoing rollup, you can set :code:`SET LOCAL citus.all_modifications_commutative TO on;`. This avoids strict locking and provides better throughput.
```

Comment:
If the replication factor defaults to more than one in Citus 6.0 then we should probably include this paragraph but remove it in a later version where that is no longer the case.